src/Serialization/upgrades/0.11.3.jl

```diff
@@ -0,0 +1,120 @@
+################################################################################
+# Upgrade Summary
+# Types of the form `Vector{T}` are only serialized with backref when
+# `is_basic_serialization_type(T) == false`. If
+# `is_basic_serialization_type(T) == true` the type `Vector{T}` does not
+# serialize `T` in the entries, but instead as a part of the `Vector` serialization.
+# Instances of the form `Tuple{T}` with `is_basic_serialization_type(T) == true` 
+# are no longer serialized with backref, however these types were already storing
+# there types outside of the JSON array of entries
+
+push!(upgrade_scripts, UpgradeScript(
+    v"0.11.3", # version this script upgrades to
+    function (s::DeserializerState, dict::Dict)
+        # file comes from polymake
+        if haskey(dict, :_ns)
+            haskey(dict[:_ns], :polymake) && return dict
+        end
+
+        # moves down tree to point where type exists in dict
+        # since we are only doing updates based on certain types
+        if !haskey(dict, :type)
+            upgraded_dict = Dict{Symbol, Any}()
+            for (key, value) in dict
+                if value isa String || value isa Vector{String} || key == :field_types
+                    upgraded_dict[key] = value
+                else
+                    if typeof(value) <: Dict{Symbol, Any}
+                        # recursive call unnamed function with var"#self"
+                        upgraded_dict[key] = var"#self#"(s, value)
+                    else
+                        values = []
+                        for v in value
+                            push!(values, var"#self#"(s, v))
+                        end
+                        upgraded_dict[key] = values
+                    end
+                end
+            end
+            return upgraded_dict
+        end
+
+        if dict[:type] == string(backref_sym)
+            backref = s.objs[UUID(dict[:id])]
+            backref_type = decodeType(backref[:type])
+
+            # recursive call unnamed function with var"#self"
+            # replace backrefs for types that no longer support backrefs
+            is_basic_serialization_type(backref_type) && return var"#self#"(s, backref)
+            return dict
+        end
+
+        # this is one of the base cases
+        !haskey(dict, :data) && return dict
+
+        # load any non backrefs into state
+        if (haskey(dict, :id))
+            s.objs[UUID(dict[:id])] = dict
```

Comment:
I've added some comments hopefully this makes things clearer otherwise we can have a call this week?

what was going on here was that i was updating the backref when it was called for, instead of storing the updated version, i pushed some changes where i store the updated version, which is more efficient, i did have to move somethings around, the other thing was that i didn't defined a way for the function to recurse on vector types so i had to just write it in the function, thats why i had to have another line where i store vector entries into state as well. I have also update this part to store the updated reference.